test/wasm/assets/016_with.yaml

```diff
@@ -229,4 +229,24 @@ cases:
   - note: with conflict
     query: |
       input = x with input.foo as 1 with input.foo.bar as 2
-    want_error: with target conflict
\ No newline at end of file
+    want_error: with target conflict
+  - note: with virtual doc iteration
+    query: |
+      x := sort([ y | y = data[_][_] ]) with data.bar.p as 3 with data.bar.q as 4; z = data.bar.p
```

Comment:
@tsandall any suggestions on this test case? I ran into a problem with getting the order back on the results if we used something more like `x := data[_][_]  with ... `, I didn't see anywhere that we sort the resultset but might have missed something. As-is its not quite the same workflow as the orginal issue (since the iteration happens inside the body of the comprehension) but I was struggling a little bit to configure one of these cases where I was confident about its flakiness.